src/domain/synthetics/markets/useMarketsInfoRequest.ts

```diff
@@ -138,274 +133,264 @@ export function useMarketsInfoRequest(chainId: number): MarketsInfoResult {
-              ["bytes32", "address", "bool"],
-              [SWAP_IMPACT_FACTOR_KEY, marketAddress, true],
-            ],
-            swapImpactFactorNegative: [
-              ["bytes32", "address", "bool"],
-              [SWAP_IMPACT_FACTOR_KEY, marketAddress, false],
-            ],
-            swapImpactExponentFactor: [
-              ["bytes32", "address"],
-              [SWAP_IMPACT_EXPONENT_FACTOR_KEY, marketAddress],
-            ],
-            longInterestUsingLongToken: [
-              ["bytes32", "address", "address", "bool"],
-              [OPEN_INTEREST_KEY, marketAddress, market.longTokenAddress, true],
-            ],
-            longInterestUsingShortToken: [
-              ["bytes32", "address", "address", "bool"],
-              [OPEN_INTEREST_KEY, marketAddress, market.shortTokenAddress, true],
-            ],
-            shortInterestUsingLongToken: [
-              ["bytes32", "address", "address", "bool"],
-              [OPEN_INTEREST_KEY, marketAddress, market.longTokenAddress, false],
-            ],
-            shortInterestUsingShortToken: [
-              ["bytes32", "address", "address", "bool"],
-              [OPEN_INTEREST_KEY, marketAddress, market.shortTokenAddress, false],
-            ],
-            longInterestInTokensUsingLongToken: [
-              ["bytes32", "address", "address", "bool"],
-              [OPEN_INTEREST_IN_TOKENS_KEY, marketAddress, market.longTokenAddress, true],
-            ],
-            longInterestInTokensUsingShortToken: [
-              ["bytes32", "address", "address", "bool"],
-              [OPEN_INTEREST_IN_TOKENS_KEY, marketAddress, market.shortTokenAddress, true],
-            ],
-            shortInterestInTokensUsingLongToken: [
-              ["bytes32", "address", "address", "bool"],
-              [OPEN_INTEREST_IN_TOKENS_KEY, marketAddress, market.longTokenAddress, false],
-            ],
-            shortInterestInTokensUsingShortToken: [
-              ["bytes32", "address", "address", "bool"],
-              [OPEN_INTEREST_IN_TOKENS_KEY, marketAddress, market.shortTokenAddress, false],
-            ],
-            virtualMarketId: [
-              ["bytes32", "address"],
-              [VIRTUAL_MARKET_ID_KEY, marketAddress],
-            ],
-            virtualLongTokenId: [
-              ["bytes32", "address"],
-              [VIRTUAL_TOKEN_ID_KEY, market.longTokenAddress],
-            ],
-            virtualShortTokenId: [
-              ["bytes32", "address"],
-              [VIRTUAL_TOKEN_ID_KEY, market.shortTokenAddress],
-            ],
-          },
-          id
-        );
-        performance.mark("useMarketsInfo-request-build-market-hash-end");
-        performance.measure(
-          "useMarketsInfo-request-build-market-hash",
-          "useMarketsInfo-request-build-market-hash-start",
-          "useMarketsInfo-request-build-market-hash-end"
-        );
+        const hashedKeys = await hashDataMapAsync({
```

Comment:
needs a fallback here as well I think. maybe we can do a fallback right inside this function (e.x. if worker not responding for 1 second or failed we resolve promise syncronously with calculated hashes)

Reply:
i agreee, added fallbacks right inside worker functons to lessen the burden of worker-function users